### PR TITLE
Feature/ci

### LINKED ISSUE #9: Create a CI to automate actions
**Is your feature request related to a problem? Please describe.**
Developers currently have to manually trigger various actions in our workflow, leading to inefficiencies and potential errors in the process.

**Describe the solution you'd like**
I would like to create a Continuous Integration (CI) setup to automate common actions in our workflow. This would include tasks such as running tests, building the application, and deploying to staging environments.

**Describe alternatives you've considered**
One alternative is to continue with manual triggering of these actions, but this is time-consuming and prone to human error. Another option is to use existing CI tools like Jenkins or GitLab CI, but setting up a custom solution might offer more flexibility for our specific needs.

**Additional context**
Our project is growing, and we need a more streamlined and error-resistant process for our development workflow. By creating a CI pipeline, we aim to improve efficiency, reduce manual errors, and ensure consistent results across our deployments.

cf : https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs

### PR DESCRIPTION
**Pull request by STR4Z**
Added Node.JS CI 

closes #9

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ "main" ]
```
Comment (TheRealHoko) [inline .github/workflows/node.js.yml:8]: development branch too

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  pull_request:
+    branches: [ "main" ]
```
Comment (TheRealHoko) [inline .github/workflows/node.js.yml:10]: same as above

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        node-version: [14.x, 16.x, 18.x]
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
```
Comment (TheRealHoko) [inline .github/workflows/node.js.yml:28]: insert this below please:
`cache-dependency-path: '**/package-lock.json'`

**Review by h-ok-o — CHANGES_REQUESTED** [review]

**Review by h-ok-o — APPROVED** [review]
LGTM 👍

**Pull request merged**